src/core/system.c

```diff
@@ -1,6 +1,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <time.h>
```

Comment:
Yes, it doesn't build on Linux without these fixes. This was tested with gcc 5.3.1 and this depends not on the compiler but on glibc which is different on Linux. In fact, it could be even insufficient on any other OS. I can't test it on Windows, so I don't know if this breaks anything there.